agent/apps/ai_guardrails/app_code/permissions.py

```
@@ -0,0 +1,29 @@
+    فقط کاربر لاگین‌کرده اجازه دسترسی داشته باشه.
+    """
+    def has_permission(self, request, view) -> bool:
+        user = getattr(request, "user", None)
+        return bool(user and getattr(user, "is_authenticated", False))
+
+
+class IsGuardrailsAdminOrReadOnly(BasePermission):
+    """
+    GET/HEAD/OPTIONS آزاد، بقیه فقط برای ادمین.
+    """
+    def has_permission(self, request, view) -> bool:
```

Ruff (0.12.2)

Missing type annotation for function argument `request`

(ANN001)